src/main/java/dune/task/TaskList.java

```diff
@@ -0,0 +1,190 @@
+package dune.task;
+
+import dune.task.Deadline;
+import dune.task.Event;
+import dune.task.Task;
+import dune.task.ToDo;
+import dune.Storage;
+import dune.DuneException;
+
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.ArrayList;
+
+/**
+ * Represents a list of tasks. A TaskList has a list of tasks.
+ */
+public class TaskList {
+    private List<Task> tasks;
+
+    private static String before = "Start date must be before end date";
+
+    /**
+     * Constructor for TaskList.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Finds tasks that contain the given string.
+     *
+     * @param string The string to search for.
+     */
+    public void find(String string) {
+        List<Task> found = new ArrayList<>();
+        for (Task t : this.tasks) {
+            if (t.getDescription().contains(string)) {
+                found.add(t);
+            }
+        }
+        if (found.size() == 0) {
+            System.out.println("No tasks found with that description");
+        } else {
+            System.out.println("Here are the matching tasks in your list:");
+            for (int i = 0; i < found.size(); i++) {
+                System.out.println(i + 1 + "." + found.get(i));
+            }
+        }
+    }
+
+    /**
+     * Adds task to list of tasks.
+     *
+     * @param task
+     */
+    public void addTask(Task task) {
+        this.tasks.add(task);
+    }
+
+    /**
+     * Adds task to list of tasks given a string.
+     *
+     * @param i  Type of task.
+     * @param text  Description of task.
+     * @param storage  Storage object to save tasks.
+     */
+    public void addTask(int i, String text, Storage storage) {
+        Task x = null;
```

Comment:
Perhaps use a more detailed name (e.g. = taskToBeAdded) (since the variable is used throughout the method (not just used on a few lines)?